### ISSUE TITLE
Facetting on numerical ranges doesn't seem to work when group_by is used

### ISSUE THREAD

**Issue by vamshiaruru32** [typesense/typesense#1209]
## Description

I've been trying out the newly released numerical range facets. It works fine normally, but when using a group_by clause, the counts are returned as zero for every facet.

## Steps to reproduce

```shell
export TYPESENSE_API_KEY=dummyAdminKeyToBeChanged

curl "http://localhost:8108/debug" \
       -H "X-TYPESENSE-API-KEY: ${TYPESENSE_API_KEY}"


curl "http://localhost:8108/collections" \
       -X POST \
       -H "Content-Type: application/json" \
       -H "X-TYPESENSE-API-KEY: ${TYPESENSE_API_KEY}" \
       -d '{
         "name": "product_variants",
         "fields": [
           {"name": "discountPercentage", "type": "float", "facet": true },
           {"name": "productSlug", "type": "string", "facet": true },
           {"name": "variantID", "type": "int32"}
         ]
       }'

curl "http://localhost:8108/collections/product_variants/documents/import?action=create" \
        -H "X-TYPESENSE-API-KEY: ${TYPESENSE_API_KEY}" \
        -H "Content-Type: text/plain" \
        -X POST \
        -d '{"id": "124","discountPercentage": 10,"productSlug": "first","variantID": 1}
            {"id": "125","discountPercentage": 20,"productSlug": "first","variantID": 2}
            {"id": "126","discountPercentage": 30,"productSlug": "second","variantID": 3}
            {"id": "127","discountPercentage": 25,"productSlug": "second","variantID": 4}
            {"id": "128","discountPercentage": 60,"productSlug": "third","variantID": 5}
            {"id": "129","discountPercentage": 70,"productSlug": "third","variantID": 6}'


# without group_by
curl --location --globoff 'http://localhost:8108/collections/product_variants/documents/search?q=*&query_by=productSlug&per_page=0&facet_by=discountPercentage(average%3A[0%2C30]%2Cgood%3A[30%2C70]%2Cgreat%3A[70%2C100])' \
--header "x-typesense-api-key: ${TYPESENSE_API_KEY}"

# with group_by
curl --location --globoff 'http://localhost:8108/collections/product_variants/documents/search?q=*&query_by=productSlug&per_page=0&group_by=productSlug&facet_by=discountPercentage(average%3A[0%2C30]%2Cgood%3A[30%2C70]%2Cgreat%3A[70%2C100])' \
--header "x-typesense-api-key: ${TYPESENSE_API_KEY}"


```
Running the last two curl commands should reproduce this behaviour.

## Expected Behavior

The facet counts are populated properly

## Actual Behavior

The facet counts are not populated properly

## Metadata

**Typesense Version**:  0.25.0

**OS**: MacOS

**Comment by vamshiaruru32** [comment]
Hi guys, is this still on track for v0.26? If yes, when is v0.26 expected to come? We are currently blocked on this feature for some of our product offerings. Thanks.

**Comment by kishorenc** [comment]
This is already fixed in recent 0.26 RC builds @vamshiaruru32 

We're in code freeze for 0.26 and are documenting things for release.

**Comment by jasonbosco** [comment]
Released in v26
